doc/manual/modules.rst

```diff
@@ -257,24 +257,32 @@ the statements in a module often involves compiling a large amount of code.
 Julia provides the ability to create precompiled versions of modules
 to reduce this time.
 
-There are two mechanisms that can achieve this:
-incremental compile and custom system image.
-
-To create a custom system image that can be used when starting Julia with the ``-J`` option,
-recompile Julia after modifying the file ``base/userimg.jl`` to require the desired modules.
-
 To create an incremental precompiled module file, add
-``__precompile__()`` at the top of your module file (before the
-``module`` starts).  This will cause it to be automatically compiled
-the first time it is imported.  Alternatively, you can manually call
-``Base.compilecache(modulename)``.  The resulting cache files will be
-stored in ``Base.LOAD_CACHE_PATH[1]``.  Subsequently, the module is
-automatically recompiled upon ``import`` whenever any of its
-dependencies change; dependencies are modules it imports, the Julia
-build, files it includes, or explicit dependencies declared by
-``include_dependency(path)`` in the module file(s).  Precompiling a
-module also recursively precompiles any modules that are imported
-therein.  If you know that it is *not* safe to precompile your module
+``__precompile__()`` at the top of your module file
+(before the ``module`` starts).
+This will cause it to be automatically compiled the first time it is imported.
+Alternatively, you can manually call ``Base.compilecache(modulename)``.
+The resulting cache files will be stored in ``Base.LOAD_CACHE_PATH[1]``.
+Subsequently, the module is automatically recompiled upon ``import``
+whenever any of its dependencies change;
+dependencies are modules it imports, the Julia build, files it includes,
+or explicit dependencies declared by ``include_dependency(path)`` in the module file(s).
+
+For file dependencies, a change is determined by examining whether the timestamp
+of each file loaded by `include` or added explicity by ``include_dependency`` is unchanged
+(to within a small margin of several seconds to account for misbehaving systems).
```

Comment:
it checks if the time exactly matches, or if the time exactly matches the rounded value. that's stricter than a margin would imply